src/kOS/Module/kOSProcessor.cs

```diff
@@ -628,12 +637,29 @@ private void ProcessElectricity(Part partObj, float time)
         {
             if (ProcessorMode == ProcessorModes.OFF) return;
 
-            RequiredPower = shared.VolumeMgr.CurrentRequiredPower;
-            var electricReq = time * RequiredPower;
-            var result = partObj.RequestResource("ElectricCharge", electricReq) / electricReq;
-
-            var newMode = (result < 0.5f) ? ProcessorModes.STARVED : ProcessorModes.READY;
-            SetMode(newMode);
+            double volumePower = 0;
+            var volume = shared.VolumeMgr.CurrentVolume;
+            if (volume.Name == "Archive")
+            {
+                // TODO: Like with the default value of ECPerBytePerSecond, this corresponds to the existing value.
+                // It should be adjusted for balance.
+                volumePower = 50000 * ECPerBytePerSecond;
+            }
+            else
+            {
+                volumePower = volume.Capacity * ECPerBytePerSecond;
+            }
+            RequiredPower = (float)(volumePower + SafeHouse.Config.InstructionsPerUpdate * ECPerInstruction / TimeWarp.fixedDeltaTime);
+            var request = volumePower * time + shared.Cpu.InstructionsThisUpdate * ECPerInstruction;
+            var available = partObj.RequestResource("ElectricCharge", request);
+            if (available / request > 0.5f)
```

Comment:
The "0.5" ratio is just what we had in there previously, and I think I've seen it used when poking around the source of other mods.  Part of me thinks that there is some practicality when dealing with time-warp conditions.  But I didn't give it much thought.